CollapseLauncher/XAMLs/MainApp/Pages/Dialogs/KeyboardShortcuts.cs

```diff
@@ -472,161 +481,249 @@ private static async void Swap_Click(object sender, RoutedEventArgs e)
+            {
+                resultList.Remove(key);
+            }
 
-                };
+            KbShortcutList = resultList;
 
-        private readonly static List<List<string>> forbiddenKeyList = new List<List<string>>
-                {
-                    new List<string> { "Ctrl", "A" },
-                    new List<string> { "Ctrl", "X" },
-                    new List<string> { "Ctrl", "C" },
-                    new List<string> { "Ctrl", "V" }
-                };
+            if (saveAfterLoad)
+                SaveKbShortcuts();
+        }
 
-        public static List<List<string>> KeyList
+        public static void SaveKbShortcuts()
         {
-            get
+            KbShortcutList ??= DefaultShortcutList;
+            string res = "";
+            foreach (var entry in KbShortcutList)
             {
-                string keyListStr = GetAppConfigValue("KbShortcutList").ToString();
-                List<List<string>> resultList = new List<List<string>>();
+                res += $"{entry.Key}:{entry.Value.ToCode()}|";
+            }
+            res = res.Remove(res.Length - 1);
+            LogWriteLine("[KeyboardShortcuts::SaveKbShortcuts] The following configuration was saved:\n\r\t" + res);
+            SetAndSaveConfigValue("KbShortcutList", res);
+        }
+        #endregion
 
-                if (keyListStr == null)
-                {
-                    foreach (List<string> list in defaultKeyList)
-                    {
-                        resultList.Add(list);
-                    }
-                    return resultList;
-                }
+        #region KbShortcut class
+        public class KbShortcut
+        {
+            public VirtualKeyModifiers Modifier { get; set; } = VirtualKeyModifiers.None;
+            public VirtualKey Key { get; set; } = VirtualKey.None;
 
-                foreach (string combination in keyListStr.Split('|'))
+            public string GetFormattedModifier()
+            {
+                return Modifier switch
                 {
-                    resultList.Add(combination.Split(",").ToList());
-                }
+                    VirtualKeyModifiers.Control => Lang._KbShortcuts.Keyboard_Control,
+                    VirtualKeyModifiers.Menu => Lang._KbShortcuts.Keyboard_Menu,
+                    VirtualKeyModifiers.Shift => Lang._KbShortcuts.Keyboard_Shift,
+                    _ => Modifier.ToString(),
+                };
+            }
 
-                if (resultList.Count < defaultKeyList.Count)
+            public string GetKey(string dictionaryKey = "")
+            {
+                return dictionaryKey switch
                 {
-                    int newKeyCount = defaultKeyList.Count - resultList.Count;
-                    resultList.InsertRange(resultList.Count, defaultKeyList.GetRange(resultList.Count, newKeyCount));
-                    KeyList = resultList;
-                }
+                    "GameSelection" => "1-3",
```

Comment:
@neon-nyan can the number of games and (max) region be dynamically retrieved here from metadata?

Reply:
Yes, the count can be retrieved from the metadata